### PR TITLE
Initial Interface properties tests

### PR DESCRIPTION
**Pull request by jashnani**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst).
- [x] New tests have been created for any new features or regression tests for bugfixes.
- [x] `tox` successfully runs, including unit tests and style checks (see [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst)).

### CODE REVIEW

File: tests/test_interface_props.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
```
Comment (epage) [inline tests/test_interface_props.py]: If you keep this as a separate file

1. Please generalize it beyond properties while scoping it to session.  For example, you could name it `test_session_intf.py` .
2. Pull in `test_intf_container` from `test_frames.py`

File: tests/test_interface_props.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        assert output_session.intf.baud_rate == 125000
+        assert output_session.intf.can_transmit_pause == True
+
+        output_session.start()
+
+        payload_list = [2, 4, 8, 16]
+        expected_frames = [
+            types.CanFrame(0, False, constants.FrameType.CAN_DATA, bytes(bytearray(payload_list)))]
+        output_session.frames.write_can(expected_frames)
+
```
Comment (epage) [inline tests/test_interface_props.py]: What value are you getting in the test by writing a can frame?

File: tests/test_interface_props.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+@pytest.mark.integration
+def test_stream_session_without_baud_rate(nixnet_out_interface):
```
Comment (epage) [inline tests/test_interface_props.py]: The recommendation I've seen (and agree with) for tests is
`test_<UNIT>_<BEHAVIOR>`.   Another way of thinking of it is that test names should serve as a description of behavior.

So maybe `test_stream_session_requires_baud_rate`

File: tests/test_interface_props.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        output_session.intf.baud_rate = 125000
+        assert output_session.intf.baud_rate == 125000
+
+        # Starting the stream session does not error because the baud_rate is set
+        output_session.start()
+
+        payload_list = [2, 4, 8, 16]
+        expected_frames = [
+            types.CanFrame(0, False, constants.FrameType.CAN_DATA, bytes(bytearray(payload_list)))]
+        output_session.frames.write_can(expected_frames)
```
Comment (epage) [inline tests/test_interface_props.py]: While we need to start the session, do we really need to write a frame?

File: tests/test_interface_props.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+
+@pytest.fixture
+def nixnet_out_interface(request):
+    interface = request.config.getoption("--nixnet-out-interface")
+    return interface
+
+
+@pytest.mark.integration
+def test_interface_props(nixnet_out_interface):
```
Comment (epage) [inline tests/test_interface_props.py]: So this is more of a `test_intf_properties_settable_only_when_stopped`?

File: tests/test_interface_props.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        output_session.stop()
+
+        # Should not error because the session has stopped.
+        output_session.intf.can_term = constants.CanTerm.OFF
+        output_session.intf.baud_rate = 125000
+        output_session.intf.can_transmit_pause = True
+
+        assert output_session.intf.can_term == constants.CanTerm.OFF
+        assert output_session.intf.baud_rate == 125000
+        assert output_session.intf.can_transmit_pause == True
```
Comment (epage) [inline tests/test_interface_props.py]: So properties not covered
- bus_err_to_in_strm
- echo_tx
- out_strm_list
- out_strm_list_by_id
- out_strm_timing
- start_trig_to_in_strm
- set_can_ext_tcvr_config
- can_lstn_only
- can_pend_tx_order
- can_sing_shot
- can_tcvr_state
- can_tcvr_type
- can_io_mode
- can_fd_baud_rate
- can_tx_io_mode
- can_fd_iso_mode
- can_edge_filter
- can_transmit_pause
- can_disable_prot_exception_handling
- src_term_start_trigger

Comment (epage) [inline tests/test_interface_props.py]: Plus all lin and flex ray (deferred out of this release)

Comment (jashnani) [inline tests/test_interface_props.py]: Are you thinking we should set all the intf properties in this test?

Comment (epage) [inline tests/test_interface_props.py]: If we're going to say we have coverage of the interface properties, then we need to test them.

Comment (jashnani) [inline tests/test_interface_props.py]: We only used a select few intf properties in this test to verify the expected start/stop behavior. We'll write a third test for all interface properties with the following checks:

- assert default 
- set the prop to non-default
- assert the new value

**Pull request merged**